File: src/main/java/ru/citeck/ecos/notifications/domain/template/dto/NotificationTemplateDto.java
package ru.citeck.ecos.notifications.domain.template.dto;

import ecos.com.fasterxml.jackson210.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.citeck.ecos.commons.data.MLText;

import java.time.Instant;
import java.util.HashMap;
import java.util.Map;

@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(value = JsonInclude.Include.NON_EMPTY)
public class NotificationTemplateDto {

    private String id;
    private String name;
    private MLText notificationTitle;

    private Map<String, TemplateDataDto> data = new HashMap<>();

    private String modifier;
    private Instant modified;
    private String creator;
    private Instant created;

    public NotificationTemplateDto(NotificationTemplateDto dto) {
        this.id = dto.id;
        this.name = dto.name;
        this.notificationTitle = dto.notificationTitle;
        this.data = dto.data;
        this.modifier = dto.modifier;
        this.modified = dto.modified;
        this.creator = dto.creator;
        this.created = dto.created;
    }

    public NotificationTemplateDto(String id) {
        this.id = id;
    }
}

File: src/main/java/ru/citeck/ecos/notifications/domain/template/dto/NotificationTemplateDto.kt
package ru.citeck.ecos.notifications.domain.template.dto

import ru.citeck.ecos.commons.data.MLText

open class NotificationTemplateDto(

    var id: String,
    var workspace: String = "",
    var name: String? = null,
    var notificationTitle: MLText? = null,
    var tags: List<String> = emptyList(),
    var model: Map<String, String>? = emptyMap(),
    var multiTemplateConfig: List<MultiTemplateElementDto>? = emptyList()

) {
    constructor(dto: NotificationTemplateDto) : this(
        dto.id,
        dto.workspace,
        dto.name,
        dto.notificationTitle,
        dto.tags,
        dto.model,
        dto.multiTemplateConfig
    )

    override fun equals(other: Any?): Boolean {

        if (this === other) {
            return true
        }
        if (javaClass != other?.javaClass) {
            return false
        }

        other as NotificationTemplateDto

        if (id != other.id ||
            workspace != other.workspace ||
            name != other.name ||
            notificationTitle != other.notificationTitle ||
            model != other.model ||
            multiTemplateConfig != other.multiTemplateConfig
        ) {

            return false
        }
        return true
    }

    override fun hashCode(): Int {
        var result = id.hashCode()
        result = 31 * result + workspace.hashCode()
        result = 31 * result + (name?.hashCode() ?: 0)
        result = 31 * result + (notificationTitle?.hashCode() ?: 0)
        result = 31 * result + (model?.hashCode() ?: 0)
        result = 31 * result + (multiTemplateConfig?.hashCode() ?: 0)
        return result
    }
}
